src/main/java/sawtooth/sdk/client/signing/Secp256k1Context.java

```diff
@@ -0,0 +1,80 @@
+package sawtooth.sdk.client.signing;
+
+import org.bitcoinj.core.ECKey;
+import org.bitcoinj.core.Sha256Hash;
+import org.bitcoinj.core.Utils;
+
+import java.security.SecureRandom;
+import java.util.logging.Logger;
+
+/**
+ * A Context for signing and verifying Secp256k1 signatures.
+ */
+public class Secp256k1Context implements Context {
+  /**
+   * Logger.
+   */
+  private static final Logger LOGGER = Logger.getLogger(Context.class.getName());
+
+  /**
+   * The number of bytes in the signature.
+   */
+  private static final int NUM_SIGNATURE_BYTES = 64;
+
+  /**
+   * Half of the number of bytes in the signature.
+   */
+  private static final int HALF_NUM_SIGNATURE_BYTES = 32;
+
+  @Override
+  public final String getAlgorithmName() {
+    return "secp256k1";
+  }
+
+  @Override
+  public final PublicKey getPublicKey(final PrivateKey privateKey) {
+    ECKey privKey = ECKey.fromPrivate(privateKey.getBytes());
+    byte[] publicKey = privKey.getPubKey();
+    return new Secp256k1PublicKey(publicKey);
+  }
+
+  /**
+   * Generate a bitcoin-style compact signature.
+   *
+   * @param privateKey ECKey private key
+   * @param data the raw message bytes
+   * @return the raw signature bytes
+   */
+  private static byte[] generateCompactSig(final ECKey privateKey, final byte[] data) {
+    Sha256Hash hash = Sha256Hash.of(data);
+    ECKey.ECDSASignature sig = privateKey.sign(hash);
+
+    byte[] csig = new byte[NUM_SIGNATURE_BYTES];
+
+    System.arraycopy(Utils.bigIntegerToBytes(sig.r, HALF_NUM_SIGNATURE_BYTES), 0,
+                                             csig, 0, HALF_NUM_SIGNATURE_BYTES);
+    System.arraycopy(Utils.bigIntegerToBytes(sig.s, HALF_NUM_SIGNATURE_BYTES), 0,
+                                             csig, HALF_NUM_SIGNATURE_BYTES, HALF_NUM_SIGNATURE_BYTES);
+    return csig;
+  }
+
+  @Override
+  public final String sign(final byte[] data, final PrivateKey privateKey) {
+    ECKey privKey = ECKey.fromPrivate(privateKey.getBytes());
+
+    return Utils.HEX.encode(generateCompactSig(privKey, data));
+  }
+
+  @Override
+  public final boolean verify(final String signature, final byte[] data, final PublicKey publicKey) {
+
```

Comment:
[nitpick] TODO comment will be nice, since we're planning to complete the feature.

Reply:
I implemented this method, so no todo comment needed.